Can't build for iOS arm64 (builtin OpenSSL error)

Trying to build for iPhone arm64 (on a Linux environment) gives error in the built-in OpenSSL. It does compile if I revert the last update (#4329). Somehow it compiles fine for iPhone armv7 architecture.

```
scons -j 1 p=iphone tools=no target=release_debug IPHONEPATH=/home/george/iostoolchain IPHONESDK=/home/george/iostoolchain/iPhoneOS9.2.sdk/ ios_triple=arm-apple-darwin11- bits=64
scons: Reading SConscript files ...
scons: done reading SConscript files.
scons: Building targets ...
/home/george/iostoolchain/usr/bin/arm-apple-darwin11-clang -o drivers/builtin_openssl2/ssl/s3_cbc.iphone.opt.debug.64.o -c -DOPUS_ARM64_OPT -fno-objc-arc -arch arm64 -fmessage-length=0 -fno-strict-aliasing -fdiagnostics-print-source-range-info -fdiagnostics-show-category=id -fdiagnostics-parseable-fixits -Wno-trigraphs -fpascal-strings -Wmissing-prototypes -Wreturn-type -Wparentheses -Wswitch -Wno-unused-parameter -Wunused-variable -Wunused-value -Wno-shorten-64-to-32 -fvisibility=hidden -Wno-sign-conversion -MMD -MT dependencies -miphoneos-version-min=5.1.1 -isysroot /home/george/iostoolchain/iPhoneOS9.2.sdk/ -Os -DNS_BLOCK_ASSERTIONS=1 -Wall -DDEBUG_ENABLED -DNEED_LONG_INT -DLIBYUV_DISABLE_NEON -fblocks -DGAME_CENTER_ENABLED -DSTOREKIT_ENABLED -DICLOUD_ENABLED -DDEBUG_MEMORY_ENABLED -DIPHONE_ENABLED -DUNIX_ENABLED -DGLES2_ENABLED -DMPC_FIXED_POINT -fexceptions -DMUSEPACK_ENABLED -DOPENSSL_ENABLED -DSQUISH_ENABLED -DVORBIS_ENABLED -DOPUS_ENABLED -DTHEORA_ENABLED -DTHEORALIB_ENABLED -DPNG_ENABLED -DDDS_ENABLED -DPVR_ENABLED -DJPG_ENABLED -DWEBP_ENABLED -DSPEEX_ENABLED -DGDSCRIPT_ENABLED -DMINIZIP_ENABLED -DXML_ENABLED -DETC1_ENABLED -DPNG_ARM_NEON_OPT=0 -DOPENSSL_NO_ASM -DOPENSSL_THREADS -DL_ENDIAN -Icore -Icore/math -Itools -Idrivers -I. -Iplatform/iphone -Iplatform/iphone/include -I/home/george/iostoolchain/iPhoneOS9.2.sdk/usr/include -I/home/george/iostoolchain/iPhoneOS9.2.sdk/System/Library/Frameworks/OpenGLES.framework/Headers -I/home/george/iostoolchain/iPhoneOS9.2.sdk/System/Library/Frameworks/AudioUnit.framework/Headers -Idrivers/builtin_openssl2 -Idrivers/builtin_zlib/zlib -Idrivers/vorbis -Idrivers/builtin_openssl2/crypto -Idrivers/builtin_openssl2/openssl -Idrivers/builtin_openssl2/crypto/evp -Idrivers/builtin_openssl2/crypto/asn1 -Idrivers/builtin_openssl2/crypto/modes drivers/builtin_openssl2/ssl/s3_cbc.c
drivers/builtin_openssl2/ssl/s3_cbc.c:459:13: error: implicit declaration of function 'private_MD5_Init' is invalid in C99 [-Werror,-Wimplicit-function-declaration,2]
         if (MD5_Init((MD5_CTX *)md_state.c) <= 0)
             ^
drivers/builtin_openssl2/openssl/md5.h:115:18: note: expanded from macro 'MD5_Init'
 #define MD5_Init private_MD5_Init
                  ^
1 error generated.
scons: *** [drivers/builtin_openssl2/ssl/s3_cbc.iphone.opt.debug.64.o] Error 1
scons: building terminated because of errors.
```

Maybe with the next update will fix this, but will keep open as a reminder.

--- a/drivers/builtin_openssl2/SCsub
+++ b/drivers/builtin_openssl2/SCsub
@@ -650,6 +650,7 @@ env_ssl.Append(CPPPATH=["#drivers/builtin_openssl2/crypto/asn1"])
 env_ssl.Append(CPPPATH=["#drivers/builtin_openssl2/crypto/modes"])
 #env_ssl.Append(CPPPATH=["#drivers/builtin_openssl2/crypto/store"])
 env_ssl.Append(CPPFLAGS=["-DOPENSSL_NO_ASM","-DOPENSSL_THREADS","-DL_ENDIAN"])
+env_ssl.Append(CFLAGS=["-Wno-error=implicit-function-declaration"]);
 
 env_ssl.add_source_files(env.drivers_sources,openssl_sources)
 

